newrelic/api/opentelemetry.py

```diff
@@ -432,30 +446,36 @@ def start_span(
                 if transaction:
                     nr_trace_type = FunctionTrace
                 elif not transaction:
-                    # This is a web request
-                    headers = self.attributes.pop("nr.http.headers", None)
-                    scheme = self.attributes.get("http.scheme")
-                    host = self.attributes.get("http.server_name")
-                    port = self.attributes.get("net.host.port")
-                    request_method = self.attributes.get("http.method")
-                    request_path = self.attributes.get("http.route")
-
-                    transaction = WebTransaction(
-                        self.nr_application,
-                        name=name,
-                        scheme=scheme,
-                        host=host,
-                        port=port,
-                        request_method=request_method,
-                        request_path=request_path,
-                        headers=headers,
-                    )
+                    if "nr.wsgi.environ" in self.attributes:
```

Comment:
This looks like the same code duplicated-do you think it would be beneficial to refactor this into a function?